ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs

```diff
@@ -41,13 +50,12 @@ chainSyncHeadersServer
        )
     => Tracer m (TraceChainSyncServerEvent blk)
     -> ChainDB m blk
+    -> Reader m blk (WithPoint blk (SerialisedHeader blk))
```

Comment:
Yes great, pass the resource in, rather than allocating it within the protocol handler.